src/db_state.rs

```diff
@@ -225,7 +226,11 @@ pub(crate) struct CoreDbState {
     pub(crate) compacted: Vec<SortedRun>,
     pub(crate) next_wal_sst_id: u64,
     pub(crate) last_compacted_wal_sst_id: u64,
-    pub(crate) last_clock_tick: i64,
+    /// The last durable clock tick is the max tick seen in any durable
+    /// SST. In cases where the WAL is enabled, that would be the max clock
+    /// tick in `next_wal_sst_id`. If WAL is disabled, it would be the max
+    /// clock tick in the most recent L0 SST.
+    pub(crate) last_durable_clock_tick: i64,
```

Comment:
I'm a little confused by this comment. What I'm seeing in `replay_wal` is that, when `replay_wal` finishes, `last_durable_clock_tick` remains as it was in the last manifest, but there could actually be durable WAL SSTs that have ticks ahead of `last_durable_clock_tick`. Seems related to this comment, too: https://github.com/slatedb/slatedb/pull/432#issuecomment-2576473357.

Reply:
Your understanding is correct, I neglected to update this comment from a previous strategy (I was actually updating the core db state at the end of replay_wal instead of just setting the tick). I'll update the comment.

I am still struggling to come up with a better name for this. This will hold the max tick up until `next_wal_sst_id - 1`, because it's possible that future WALs have been flushed that the latest manifest does not yet know about. I could name it `last_known_durable_clock_tick` but that feels a bit verbose 😓

Thoughts?